gems/smithy-schema/lib/smithy-schema/document/serializer.rb

```diff
@@ -0,0 +1,283 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'time'
+
+module Smithy
+  module Schema
+    module Document
+      # Serializes data into a document data.
+      class Serializer
```

Comment:
Should this really be a public API? I would think these classes should be private behind some methods in Document.